libraries/chain/database.cpp

```diff
@@ -1723,30 +1723,40 @@ void database::clear_null_account_balance()
       adjust_supply( -total_sbd );
 }
 
+void update_children_rshares2( database& db, const comment_object& c, const fc::uint128_t& old_rshares2, const fc::uint128_t& new_rshares2 )
+{
+   // Updates the children_rshares2 field of the comment and all its ancestors
+
+   const comment_object* current_comment = &c;
+   while( true )
+   {
+      db.modify( *current_comment, [&]( comment_object& comment )
+      {
+         comment.children_rshares2 -= old_rshares2;
+         comment.children_rshares2 += new_rshares2;
+      } );
+
+      if( current_comment->depth == 0 )
+         break;
+
+      current_comment = &db.get_comment( current_comment->parent_author, current_comment->parent_permlink );
+   }
+}
+
 /**
```

Comment:
This comment is not accurate. It wasn't before this change, but it certainly is not accurate now.

Updating rshares2 for a single comment is needed for payout calculations.
Updating children_rshares2 is non-consensus and could be skipped for validation only nodes.